osbrain/tests/test_agent_async_requests.py

```diff
@@ -139,63 +124,73 @@ def test_unknown(nsproxy, server_client_late_reply_return):
 
     # Manually remove the pending request before it is received
     client.set_attr(_pending_requests={})
-    time.sleep(1.1)
-    assert server.get_attr('received') == ['foo']
-    assert client.get_attr('received') == []
+    server.unsafe.set_attr(blocked=False)
+    assert wait_agent_attr(server, value=['foo'], timeout=.1)
+    assert not wait_agent_attr(client, length=1, timeout=.5)
     assert logger_received(logger,
                            log_name='log_history_warning',
                            message='Received response for an unknown request!')
 
 
-def test_wait(nsproxy, server_client_late_reply_delay):
+def test_wait_in_time(nsproxy):
     """
     Asynchronous request-reply pattern maximum wait.
     """
-    server, client = server_client_late_reply_delay
+    client, server = async_client_server()
 
     logger = run_logger('logger')
     client.set_logger(logger)
     sync_agent_logger(client, logger)
 
     # Response received in time
-    client.send('async', 1, wait=2)
-    time.sleep(1.1)
-    assert server.get_attr('received') == [1]
-    assert client.get_attr('received') == ['x1']
+    client.send('async', '1', wait=2)
+    time.sleep(1)
+    server.unsafe.set_attr(blocked=False)
+    assert wait_agent_attr(server, value=['1'], timeout=.1)
+    assert wait_agent_attr(client, value=['x1'], timeout=.1)
```

Comment:
Maybe we could:

```python
client.send(...)
assert wait_agent_attr(server, ...)
server.unsafe.set_attr(...)
assert wait_agent_attr(client, ...)
```

That way we remove the sleep all together and make the test a bit faster.

Reply:
Okay about removing the timeouts.

Reordering is important to avoid (very unlikely) situation:

```python
client.send(...)
server.unsafe.set_attr(blocked=False)  # The server did not receive the message yet,
                                       # so it will never un-block?
```